flexget/plugins/input/telegram_input.py

```diff
@@ -0,0 +1,228 @@
+            entry = Entry()
+            entry['title'] =  text
+
+            #We need a url, so we add a dummy
+            entry['url'] = "http://localhost?update_id="+str(update_id)
+
+            #Store the message if we need to use it in other plugins
+            entry['telegram_message'] =  message['message']
+            
+            #Check Types
+            if types:
+                if message['message']['chat']['type'] not in types:
+                    logger.debug("Ignoring message because of invalid type {}",message)
+                    continue
+
+            #Check From
+            append = True
+            if whitelist:
+                append = False
+                for i in whitelist:
+                    if 'username' in i and i['username'] == message['message']['from']['username']:
+                        logger.debug("WhiteListing: Username {}",message['message']['from']['username'])
+                        append = True
+                        break
+                    elif ('fullname' in i) and (i ['fullname']['first'] == message['message']['from']['first_name'] and i['fullname']['sur'] == message['message']['from']['last_name']):
+                        logger.debug("WhiteListing: Full Name {} {}",message['message']['from']['first_name'],message['message']['from']['last_name'])
+                        append = True
+                        break
+                    elif 'group' in i:
+                        if message['message']['chat']['type'] == 'group':
+                            if message['message']['chat']['title'] == i['group']:
+                                logger.debug("WhiteListing: Group {}",message['message']['chat']['title'])
+                                append = True
+                                break
+
+                if not append:
+                    logger.debug("Ignoring message because of no whitelist match {}",message)
+                    continue
+            
+            #Process the entry config
+            if not entry_config:
+                pass
+            else:
+                for field, regexp in entry_config.items():
+                    match = re.search(regexp, text)
+                    if match:
+                        try:
+                            # add field to entry
+                            entry[field] = match.group(1)
+                        except IndexError:
+                            logger.error('Regex for field `{}` must contain a capture group', field)
+                            raise plugin.PluginError(
+                                'Your telegram_input plugin config contains errors, please correct them.'
+                            )
+                    else:
+                        logger.debug('Ignored entry, not match on field {}: {}',field, entry)
+                        append = False
+                        break
+            
+            #Append the entry
+            if append:
```

Comment:
please use the `for/else` pattern instead of a control variable

Reply:
I can't use the else here because I need to run the complete loop break it if any check fails! The else in this case will make the code more complicated, because I would need to append the entry in the else of the for, and then in the case of no entry_config is informed